app/configurator/interactive-filters/time-slider.tsx

```diff
@@ -0,0 +1,61 @@
+import { scaleLinear } from "d3";
+import React, { ChangeEvent } from "react";
+
+import { ChartState, useChartState } from "@/charts/shared/use-chart-state";
+import { useInteractiveFilters } from "@/charts/shared/use-interactive-filters";
+import { TableChartState } from "@/charts/table/table-state";
+import { Slider } from "@/components/form";
+import { parseDate } from "@/configurator/components/ui-helpers";
+
+export const TimeSlider = ({ componentIri }: { componentIri?: string }) => {
+  const [_, dispatch] = useInteractiveFilters();
+  const [t, setT] = React.useState(0);
+  const chartState = useChartState() as NonNullable<
+    Exclude<ChartState, TableChartState>
+  >;
+  const onChange = React.useMemo(() => {
+    // FIXME: enable interactive filters for maps!
+    if (componentIri && chartState.chartType !== "map") {
+      const rawRange = [
+        ...new Set(
+          chartState.allData.map((d) => d[componentIri]).filter(Boolean)
+        ),
+      ].sort() as string[];
+      const range = rawRange.map(parseDate);
+      const [min, max] = [range[0], range[range.length - 1]];
+      const scale = scaleLinear().range([min.getTime(), max.getTime()]);
+      const reversedRange = range.reverse();
```

Comment:
The computation of the available values for the range can be done once outside of the onChange. I think d3 extent could be used here to get the min/max without sorting.